src/libs/libRocket/Include/Rocket/Core/StringBase.inl

```diff
@@ -165,12 +172,18 @@ void StringBase< T >::Reserve(size_type size)
 	if (value == (T*)local_buffer)
 	{
 		T* new_value = (T*)realloc(NULL, buffer_size);
+		// TODO! Consider throwing here.
```

Comment:
I'll discuss this more on irc and comply based on the outcome of that. But the current argument  makes no sense to me.

Is there anywhere that it says librocket shouldn't use exceptions for any reason? If there is none, perhaps we should add some and the justification for it?

But given the rest of stl does use exceptions it seems exceptions will happen somewhere anyway?

Also given there are only two places to add the bad_alloc support to address the issue properly, it seems deficient not to address the issue unless there is good reason. I mean why force any user of this library to crash with no message or a misleading access violation one if there is no good reason, when we can instead with trivial effort leave it up to the app to decide to crash or not by simply catching the exception or not and without a misleading message. That's what they are for, right?

To do otherwise without good reason seems short sited and illogical unless a real "customer" of the library explicitly needs exceptions disabled and even then it should probably a a compile time enabled thing.

In summary, I think the argument for not using exceptions is insufficiently justified so far to agree with.